### PR TITLE
Remove staging-areas module

### PR DESCRIPTION
**Pull request by bbpennel**
Removes staging areas module from usage in the CDR, adds new but significantly reduced staging policy classes to deposit app, updates deposit jobs and tests accordingly, adds missing unit tests for deposit jobs that involved staging.

### CODE REVIEW

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java [hunk 1]
```diff
@@ -103,35 +113,23 @@ public void runJob() {
...
 					break;
 				case DELETE_INGESTED_FILES:
-					deleteFile(storageUri);
+					deleteFile(stagingUri);
 					break;
 				case DELETE_INGESTED_FILES_EMPTY_FOLDERS:
-					File parent = deleteFile(storageUri);
+					File parent = deleteFile(stagingUri);
 					if (parent != null && parent.exists()) {
 						if (parent.list().length == 0) {
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Is there some crazy indenting that starts here, or is it just me?

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java [hunk 2]
```diff
@@ -155,35 +153,23 @@ private void deleteStagedFiles(Model m) {
...
+
 	// Cleanup files and directories specifically requested be cleaned up by an earlier job
 	private void deleteCleanupFiles(Model m) {
 		List<String> cleanupPaths = new ArrayList<>();
-		
+
 		// Create a list of files that need to be cleaned up
 		NodeIterator it = m.listObjectsOfProperty(CdrDeposit.cleanupLocation);
 		try {
 			while (it.hasNext()) {
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: More weird indenting

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java [hunk 3]
```diff
@@ -155,35 +153,23 @@ private void deleteStagedFiles(Model m) {
...
-				} catch (StagingException e) {
-					LOG.error("Could not resolve storage URI: {}", cleanupUri.toString(), e);
-				}
-				
-				CleanupPolicy p = area.getIngestCleanupPolicy();
-				switch (p) {
+
+				CleanupPolicy cleanupPolicy = stagingManager.getCleanupPolicy(cleanupUri);
+				switch (cleanupPolicy) {
 				case DELETE_INGESTED_FILES:
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Is this case simply intended to fall through to the DELETE_INGESTED_FILES_EMPTY_FOLDERS case?

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java [hunk 4]
```diff
@@ -215,32 +201,16 @@ private void deleteCleanupFiles(Model m) {
...
 
-	public Stages getStages() {
-		return stages;
+	public void setStatusKeysExpireSeconds(int seconds) {
+		this.statusKeysExpireSeconds = seconds;
 	}
 
-	public void setStages(Stages stages) {
-		this.stages = stages;
+	public StagingPolicyManager getStagingManager() {
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Could we rename this method "getStagingPolicyManager()" just to make it clearer what it's returning?

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java [hunk 5]
```diff
@@ -215,32 +201,16 @@ private void deleteCleanupFiles(Model m) {
...
+		return stagingManager;
 	}
 
-	public void setStatusKeysExpireSeconds(int seconds) {
-		this.statusKeysExpireSeconds = seconds;
+	public void setStagingManager(StagingPolicyManager stagingManager) {
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Ditto on renaming

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java (hunk 1 above)
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Eclipse's indentation for switch blocks is kind of weird in my opinion, but i opted not to fight it.  Or are you mainly referring to the fact that the code gets pretty far indented?

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java (hunk 3 above)
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Yes, that is how it works.  Even though its not part of this PR, maybe I should switch to an if statement since there is only a single case.

File: deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java (hunk 1 above)
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/CleanupDepositJob.java]: Yeah, more the latter. It appears that there are double indents at each level, which is pushing the code out a a standard page width.

File: deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java [hunk 6]
```diff
@@ -18,84 +30,64 @@
...
-import edu.unc.lib.dl.fedora.PID;
 import edu.unc.lib.dl.rdf.CdrDeposit;
-import edu.unc.lib.staging.Stages;
-import edu.unc.lib.staging.StagingException;
 
+/**
+ * Verifies that files referenced by this deposit for ingest are present and
+ * available
+ * 
+ * @author daines
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java]: Do we need this authorship attribution?

File: deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java [hunk 7]
```diff
@@ -18,84 +30,64 @@
...
+		// Construct a map of objects to file paths to verify
+		Set<String> hrefs = new HashSet<String>();
+		addLocations(hrefs, model, CdrDeposit.stagingLocation);
+		addLocations(hrefs, model, CdrDeposit.hasDatastream);
 
 		setTotalClicks(hrefs.size());
 
-		for (Entry<PID, String> href : hrefs.entrySet()) {
-			verifyRunning();
+		// Verify that the deposit is still running before preceeding with check
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java]: "proceeding"

File: deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java [hunk 8]
```diff
@@ -104,4 +96,13 @@ public void runJob() {
 		}
 	}
 
+	private void addLocations(Set<String> hrefs, Model model, Property property) {
+		Property fileLocation = CdrDeposit.stagingLocation;
+		StmtIterator i = model.listStatements(new SimpleSelector((Resource) null, fileLocation, (RDFNode) null));
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java]: Do nulls really need casts for this to work?

File: deposit/src/main/java/edu/unc/lib/deposit/staging/StagingPolicyManager.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	/**
+	 * Returns true if the given file uri is contained by at least one of the
+	 * staging locations registered with this manager
+	 * 
+	 * @param fileUri
+	 * @return
+	 */
+	public boolean inValidStagingLocation(URI fileUri) {
```
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/staging/StagingPolicyManager.java]: yeah that is pretty unfortunate isn't it.  I think i'll change it to isValidStagingLocation

File: deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java (hunk 8 above)
Comment (bbpennel) [inline deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java]: yeah, othewise it won't compile since it can't determine which overridden version of the method to use

File: deposit/src/test/java/edu/unc/lib/deposit/staging/StagingPolicyManagerTest.java [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+	@Test(expected = StagingException.class)
+	public void getCleanupPolicyForUnregistedPathTest() {
+		manager.init();
+
+		manager.getCleanupPolicy(unregisteredUri);
+	}
+
+	@Test
+	public void inValidStagingLocationTest() {
```
Comment (harringj) [inline deposit/src/test/java/edu/unc/lib/deposit/staging/StagingPolicyManagerTest.java]: Same issue with method name as above

File: deposit/src/main/java/edu/unc/lib/deposit/staging/StagingPolicyManager.java [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+					.constructCollectionType(List.class, StagingPolicy.class));
+		} catch (IOException e) {
+			throw new StagingException("Failed to load staging configuration from " + configPath, e);
+		}
+
+		// Verify and resolve any staging location paths
+		for (StagingPolicy policy : policies) {
+			String path = policy.getPath();
+
+			// If path was no absolute, resolve it against the base path
```
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/staging/StagingPolicyManager.java]: "not"

File: deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java (hunk 8 above)
Comment (harringj) [inline deposit/src/main/java/edu/unc/lib/deposit/validate/ValidateFileAvailabilityJob.java]: Ahh, that makes sense
